offline_cam/data_loading.hpp

```diff
@@ -0,0 +1,82 @@
+#include <map>
+#include <fstream>
+#include <string>
+#include <optional>
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+#include <eigen3/Eigen/Dense>
+
+#include "csv_iterator.hpp"
+
+typedef unsigned long long ullong;
+
+//#define LAZY
```

Comment:
Let's document this:

Uncommenting this preprocessor macro makes the offline_cam load each data from the disk as it is needed. Otherwise, we load all of them at the beginning, hold them in memory, and drop them in the queue as needed. Lazy loading has an artificial negative impact on performance which is absent from an online-sensor system. Eager loading deteriorates the startup time and uses more memory. I prefer eager loading.

Reply:
updated.